spec/service_spec.rb

```diff
@@ -48,6 +48,47 @@
     expect { service.resource_get('api') }.to raise_error(Exception, "BLAH")
   end
 
+  context "#get_resource_by_ems_ref" do
+    it "fetches data_center" do
+      return_message = <<-EOX.chomp
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<data_center href="/api/datacenters/00000001-0001-0001-0001-0000000000f1" id="00000001-0001-0001-0001-0000000000f1">
+    <name>Default</name>
+    <description>The default Data Center</description>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/storagedomains" rel="storagedomains"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/clusters" rel="clusters"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/networks" rel="networks"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/permissions" rel="permissions"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/quotas" rel="quotas"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/qoss" rel="qoss"/>
+    <link href="/api/datacenters/00000001-0001-0001-0001-0000000000f1/iscsibonds" rel="iscsibonds"/>
+    <local>false</local>
+    <storage_format>v3</storage_format>
+    <version major="3" minor="6"/>
+    <supported_versions>
+        <version major="3" minor="6"/>
+    </supported_versions>
+    <status>
+        <state>up</state>
+    </status>
+    <mac_pool href="/api/macpools/0000000d-000d-000d-000d-00000000037a" id="0000000d-000d-000d-000d-00000000037a"/>
+    <quota_mode>disabled</quota_mode>
+</data_center>
+EOX
+      expect(service).to receive(:resource_get).and_return(return_message)
+      expect(service.class).to receive(:name_to_class).with("data_center").and_return(Ovirt::DataCenter)
```

Comment:
Sorry, I should have been more clear that I want to get rid of this stub.  The right XML response from the `resource_get` call should fix the need to stub `name_to_class`.

We are testing against different versions, but the XML structure should be the same.  You can `puts #{xml.to_s}` after https://github.com/ManageIQ/ovirt/pull/50/files#diff-9be2dde14cfe2ea852f52a1c2bc13398R101 to verify that we see the same structure.
If they are different, we're going to have a lot more work to do to support v3.6 :disappointed: